client/src/components/primitiveComponents/TimelineTable.tsx

```diff
@@ -0,0 +1,79 @@
+import { Button, Heading, Flex, Separator, Box, Table } from '@radix-ui/themes';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { TLAddTimelineAlert, TLEditTimelineAlert } from './AlertBoxes';
+
+interface TimelineProps {
+    timeline: Record<number, string[]>;
+    onChange: Function;
```

Comment:
Should probably specify the input and outputs of this type to actually benefit from Typescript here:
`(timeline: Record<number, string[]) => void`

Reply:
i believe i got them all